extmod/vfs.c

```diff
@@ -164,23 +164,25 @@ STATIC mp_obj_t mp_vfs_autodetect(mp_obj_t bdev_obj) {
         mp_vfs_blockdev_t blockdev;
         mp_vfs_blockdev_init(&blockdev, bdev_obj);
         uint8_t buf[44];
-        mp_vfs_blockdev_read_ext(&blockdev, 0, 8, sizeof(buf), buf);
-        #if MICROPY_VFS_LFS1
-        if (memcmp(&buf[32], "littlefs", 8) == 0) {
-            // LFS1
-            vfs = mp_type_vfs_lfs1.make_new(&mp_type_vfs_lfs1, 1, 0, &bdev_obj);
-            nlr_pop();
-            return vfs;
-        }
-        #endif
-        #if MICROPY_VFS_LFS2
-        if (memcmp(&buf[0], "littlefs", 8) == 0) {
-            // LFS2
-            vfs = mp_type_vfs_lfs2.make_new(&mp_type_vfs_lfs2, 1, 0, &bdev_obj);
-            nlr_pop();
-            return vfs;
+        for (size_t block_num = 0; block_num <= 1; ++block_num) {
```

Comment:
Comment explaining what this is doing:

`// The superblock for littlefs is in both block 0 and 1, but block 0 may be erased or partially written, so block 1 must be checked if block 0 does not have a valid littlefs superblock in it.`

Can there be a situation where block 1 has the "true" superblock, but block 0 appears to look valid? Or doesn't it matter in this case because they'll at least still agree on the block size?

Reply:
Here we don't care about the block size or any data in the superblock, just if it looks like a littlefs device so that we can attempt to mount it as such.

In the stm32 code however, it does use the block len and number of blocks to compute the total size of the block device.  But you are right that these values never change so should agree in both superblocks.  The assumption is that if the text "littlefs" is in the right location then the block len and number of blocks fields are valid.

Reply:
Comment added.